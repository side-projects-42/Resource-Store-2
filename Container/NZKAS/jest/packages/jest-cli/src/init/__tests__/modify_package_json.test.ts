/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import modifyPackageJson from '../modify_package_json';

test('should remove jest config if exists', () => {
  expect(
    modifyPackageJson({
      projectPackageJson: {
        jest: {
          collectCoverage: true,
        },
      },
      shouldModifyScripts: true,
    }),
  ).toMatchSnapshot();
});

test('should add test script when there are no scripts', () => {
  expect(
    modifyPackageJson({
      projectPackageJson: {},
      shouldModifyScripts: true,
    }),
  ).toMatchSnapshot();
});

test('should add test script when there are scripts', () => {
  expect(
    modifyPackageJson({
      projectPackageJson: {
        scripts: {
          lint: 'eslint .',
          test: 'jasmine',
        },
      },
      shouldModifyScripts: true,
    }),
  ).toMatchSnapshot();
});

test('should not add test script when { shouldModifyScripts: false }', () => {
  expect(
    modifyPackageJson({
      projectPackageJson: {},
      shouldModifyScripts: false,
    }),
  ).toMatchSnapshot();
});
