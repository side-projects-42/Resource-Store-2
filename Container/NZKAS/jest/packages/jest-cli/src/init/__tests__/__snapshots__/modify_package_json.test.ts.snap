// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add test script when there are no scripts 1`] = `
"{
  \\"scripts\\": {
    \\"test\\": \\"jest\\"
  }
}
"
`;

exports[`should add test script when there are scripts 1`] = `
"{
  \\"scripts\\": {
    \\"lint\\": \\"eslint .\\",
    \\"test\\": \\"jest\\"
  }
}
"
`;

exports[`should not add test script when { shouldModifyScripts: false } 1`] = `
"{}
"
`;

exports[`should remove jest config if exists 1`] = `
"{
  \\"scripts\\": {
    \\"test\\": \\"jest\\"
  }
}
"
`;
